cmd/warpforge/internal/util/util.go

```diff
@@ -94,51 +107,55 @@ func PlotFromFile(filename string) (wfapi.Plot, error) {
 //    - warpforge-error-plot-invalid -- when the plot data is invalid
 //    - warpforge-error-plot-step-failed --
 //    - warpforge-error-serialization -- when the module or plot cannot be parsed
-//    - warpforge-error-unknown -- when changing directories fails
 //    - warpforge-error-workspace -- when opening the workspace set fails
-//    - warpforge-error-datatoonew -- when module or plot data version is unrecognized
-func ExecModule(ctx context.Context, config wfapi.PlotExecConfig, fileName string) (wfapi.PlotResults, error) {
-	ctx, span := tracing.Start(ctx, "execModule")
-	defer span.End()
-	result := wfapi.PlotResults{}
+//    - warpforge-error-datatoonew -- when error is too new
+//    - warpforge-error-searching-filesystem -- unexpected error traversing filesystem
+//    - warpforge-error-path-absolute -- fail to canonicalize module path
+//    - warpforge-error-initialization -- fail to get working directory or executable path
+func ExecModule(ctx context.Context, wss workspace.WorkspaceSet, pltCfg wfapi.PlotExecConfig, fileName string) (result wfapi.PlotResults, err error) {
+	ctx, span := tracing.StartFn(ctx, "execModule")
+	defer func() { tracing.EndWithStatus(span, err) }()
 
 	fsys := os.DirFS("/")
+
 	// parse the module, even though it is not currently used
 	if _, err := dab.ModuleFromFile(fsys, fileName); err != nil {
 		return result, err
 	}
-
-	plot, werr := dab.PlotFromFile(fsys, filepath.Join(filepath.Dir(fileName), dab.MagicFilename_Plot))
-	if werr != nil {
-		return result, werr
+	modulePath := filepath.Dir(fileName)
+	modulePath, xerr := filepath.Abs(modulePath)
+	if xerr != nil {
+		return result, serum.Error(wfapi.ECodePathAbs, serum.WithCause(xerr),
```

Comment:
Gonna pick on this error code because I see it only comes up once, here: can we reuse some other existing error code for it?  Or perhaps get rid of this error branch entirely?

`filepath.Abs` is doing an `os.Getwd` internally.  That's the only possible source of its errors.  If we already have cwd somewhere, we can just do the clean+ifAbs+join ourselves and it would have no error branch.

Reply:
That's good to know. Probably a potential source of bugs in all sorts of places.

Reply:
We've been using `ECodeIO` in a bunch of places for this. I'm not a fan of that either but we can maybe phase it out.

Reply:
I think `ECodeInitialization` in some places now too.  I'd probably lean towards that one (and use ECodeIO more for things that error while in motion).  (Weakly held opinion.)